src/ceph_osd.cc

```diff
@@ -560,8 +560,16 @@ int main(int argc, const char **argv)
   ms_objecter->set_default_policy(Messenger::Policy::lossy_client(CEPH_FEATURE_OSDREPLYMUX));
 
   entity_addrvec_t public_addrs, cluster_addrs;
-  pick_addresses(g_ceph_context, CEPH_PICK_ADDRESS_PUBLIC, &public_addrs);
-  pick_addresses(g_ceph_context, CEPH_PICK_ADDRESS_CLUSTER, &cluster_addrs);
+  r = pick_addresses(g_ceph_context, CEPH_PICK_ADDRESS_PUBLIC, &public_addrs);
+  if (r < 0) {
+    derr << "Failed to pick public address." << dendl;
```

Comment:
Hmm, if you as a user see "Failed to pick...address", what does that mean to you?

I'm not sure if the error message itself can be super informative about this particular issue, but if we can't make it really clear to the user then we should probably document the error message so they can google for it? :)

Reply:
I can probably change the whole method `pick_addresses` in order to make it a little bit more approachable for better error reporting if there are no objections. It will be a bigger change than this one. Adding docs seems like the last resort.

Reply:
Docs are good! I'm not sure you can squeeze a meaningful explanation of what's happened into a reasonably-sized error message.

Reply:
(I should note it looks good other than needing something for confused users.)

Reply:
Where do I update docs?